diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 71d6427..6ba6406 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1174,7 +1174,7 @@ config("chromium_code") {
       # Non-chromium code is not guaranteed to compile cleanly with
       # _FORTIFY_SOURCE. Also, fortified build may fail when optimizations are
       # disabled, so only do that for Release build.
-      defines += [ "_FORTIFY_SOURCE=2" ]
+      # defines += [ "_FORTIFY_SOURCE=2" ]
     }
 
     if (is_mac || is_ios) {
